src/services/catalog.service.ts

```diff
@@ -145,26 +145,56 @@ export class ImageItem implements IServiceItem {
   public description: string;
   public longDescription: string;
   public resource: any;
-  public isBuilder: boolean;
-  public isHidden: boolean;
+  private validSpecTagName: any;
   private tags: any;
   private catalogSrv: CatalogService;
 
   constructor (image: any, catalogSrv : CatalogService) {
     this.resource = image;
     this.catalogSrv = catalogSrv;
-    this.tags = this.catalogSrv.$filter('imageStreamTagTags')(this.resource);
-    this.iconClass = this.getIcon();
-    this.name = this.getName();
-    this.description = this.getDescription();
-    this.longDescription = this.getLongDescription();
-    this.catsBySubCats = this.getCategoriesBySubCategories();
-    this.isBuilder = _.includes(this.tags, 'builder');
-    this.isHidden = _.includes(this.tags, 'hidden');
+    this.validSpecTagName = this.getValidSpecTagName();
+    if (this.validSpecTagName) {
+      this.tags = this.getTags();
+      this.iconClass = this.getIcon();
+      this.name = this.getName();
+      this.description = this.getDescription();
+      this.longDescription = this.getLongDescription();
+      this.catsBySubCats = this.getCategoriesBySubCategories();
```

Comment:
Ok, I was thinking the same.  IServiceItem.resource (the API object) should be passed to the ordering wizard.